src/SimpleFiber.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace React\Async;
+
+use React\EventLoop\Loop;
+
+/**
+ * @internal
+ */
+final class SimpleFiber
+{
+    private ?\Fiber $fiber = null;
+
+    public function __construct()
+    {
+        $this->fiber = \Fiber::getCurrent();
+    }
+
+    public function resume(mixed $value)
+    {
+        if ($this->fiber === null) {
+            Loop::futureTick(static fn() => \Fiber::suspend(static fn() => $value));
+            return;
+        }
+
+        Loop::futureTick(fn() => $this->fiber->resume($value));
+    }
+
+    public function throw(mixed $throwable)
+    {
+        if (!$throwable instanceof \Throwable) {
+            $throwable = new \UnexpectedValueException(
+                'Promise rejected with unexpected value of type ' . (is_object($throwable) ? get_class($throwable) : gettype($throwable))
+            );
+        }
+
+        if ($this->fiber === null) {
+            Loop::futureTick(static fn() => \Fiber::suspend(static fn() => throw $throwable));
```

Comment:
This is an issue with Revolt and an event loop adapter. Revolt runs callbacks in a separate fiber, so they're safe to suspend at any time, which might happen, because any code can be called. However, that means this suspend will suspend that extra fiber and not the event loop fiber itself.

Reply:
Will have a look tonight to see if we really need to do this. But earlier when working on this there wasn't a way around of not doing this.

Reply:
@WyriHaximus If `SimpleFiber` implemented an interface, and instances were created by a definable factory, then it would be possible to provide an adapter to Revolt I think.

Reply:
@trowski With a default to ours, but let you set it to yours to make it an adapter work, right?